File: Decorator/SaborMorango.php
<?php
require_once "./ISorvete.php";

class SaborMorango implements ISorvete{

    private $objetoDecorado;

    public function __construct(ISorvete $objetoDecorado)
    {
        $this->objetoDecorado = $objetoDecorado;
    }

    public function preco(){
        return $this->objetoDecorado->preco() + 2;
    }
}

File: Java/decorator/SaborMorango.java
package deco;

public class SaborMorango extends SorveteDecorator{

	public SaborMorango(ISorvete objetoDecorado) {
		super(objetoDecorado);
	}

	@Override
	public float preco() {
		return this.objetoDecorado.preco() + 2f;
	}

}
